src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs

```diff
@@ -69,6 +69,15 @@ public ContentDialog() : base()
 					that.UpdateVisualState();
 				}
 			};
+
+			_popup.Closed += (s, e) =>
+			{
+				if (thisRef.Target is ContentDialog that)
+				{
+					that.Hide();
+				}
```

Comment:
Do we need something similar for other controls that internally create a Popup?

Reply:
I'd guess that flyout would need the same treatment. Also, a runtime test would be needed :)

Reply:
I'm not sure we'd need this for other things such as Flyout. The problem here is because we are expecting a result to be returned from the ContentDialog once it closes. Without this logic, the `ShowAsync()` call never returns the expected `ContentDialogResult`. Whereas Flyout just has a void ShowAt, so nothing is waiting for the result of the method. And Flyout already takes action to hide itself when its Popup is closed, [here](https://github.com/unoplatform/uno/blob/e8d56677ba8af1cca929e2ded5693d69c8ab764b/src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutBase.cs#L409-L413)